reference/conanfile/tools/gnu/pkgconfigdeps.rst

```diff
@@ -0,0 +1,94 @@
+.. _conan_tools_pkgconfig:
+
+
+conan.tools.gnu.pkgconfigdeps
+=============================
+
+.. warning::
+
+    These tools are **experimental** and subject to breaking changes.
+
+
+PkgConfigDeps
+-------------
+
+Available since: `1.38.0 <https://github.com/conan-io/conan/releases>`_
+
+
+The ``PkgConfigDeps`` is the dependencies generator for pkg-config. Generates pkg-config files named *<PKG-NAME>.pc*
+(where ``<PKG-NAME`` is the name declared by dependencies in ``cpp_info.name`` or in ``cpp_info.names["pkg_config"]``
+if specified), containing a valid pkg-config file syntax. Indeed, it can also be defined using ``set_property`` and the
+property ``pkg_config_name`` (available since Conan 1.36), for instance:
+
+.. code-block:: python
+
+    self.cpp_info.components["mycomponent"].set_property("pkg_config_name", "mypkg-config-name")
+
+
+.. note::
+    In Conan 2.0 that will be the default way of setting those properties and also passing custom properties to generators.
+    Check the :ref:`cpp_info attributes reference <cpp_info_attributes_reference>` for more information.
+
+
+The ``prefix`` variable is automatically adjusted to the ``package_folder``.
+
+
+The ``PkgConfigDeps`` generator can be used by name in conanfiles:
+
+.. code-block:: python
+    :caption: conanfile.py
+
+    class Pkg(ConanFile):
+        generators = "PkgConfigDeps"
+
+.. code-block:: text
+    :caption: conanfile.txt
+
+    [generators]
+    PkgConfigDeps
+
+And it can also be fully instantiated in the conanfile ``generate()`` method:
+
+.. code:: python
+
+    from conans import ConanFile
+    from conan.tools.gnu import PkgConfigDeps
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+
+        def generate(self):
+            pc = PkgConfigDeps(self)
+            pc.generate()
+
+The ``PkgConfigDeps`` will generate after a ``conan install`` command the `*.pc`:
+
+.. code-block:: bash
+
+    $ conan install .
+    # Check the [PC_FILE_NAME].pc created in your current folder
+
+For instance, the `.pc` created could look like this one:
+
+.. code-block:: text
+
+    libdir=/my_absoulte_path/mylib/lib
+    libdir2=${prefix}/lib2
+    includedir=/my_absoulte_path/mylib/include
+
+    Name: MyLib
+    Description: Conan package: MyLib
+    Version: 0.1
+    Libs: -L"${libdir}" -L"${libdir2}"%s
+    Cflags: -I"${includedir}"
+
+
+Components
+++++++++++
+
+If a recipe uses :ref:`components<package_information_components>`, the files generated will be *<COMP-NAME>.pc* with their corresponding
```

Comment:
Isn't the name ``PKG-NAME_COMP-NAME.pc``? I think it could be better this way, to avoid potential conflicts with components of different packages. Or is not possible?

Reply:
The thing is that I got this doc-lines about the Components from this one https://docs.conan.io/en/latest/reference/generators/pkg_config.html Anyway, I'm gonna check it 😉